### PR TITLE
[robotic grounding] Add whole-body retargeting and kinematic replay

### PR DESCRIPTION
**Pull request by huihuaNvidia2023**
## Changes:

- New retarget pipeline (nvhuman_to_g1.py, whole_body_kinematics.py) with ground-plane alignment, foot-Z clamping, and object-specific lift
- Kinematic replay script (replay_motion.py) supporting both single-robot and dual-hand schemas with auto-detect for legacy vs grounded data
- Support surface reconstruction extended for NvhumanG1 schema
- Parquet replay adapter (replay_data.py) with schema auto-detection
- Scene viewer updated to support robot articulation spawning
- README updated with retargeting, replay, and reconstruction docs
- Integration and unit tests for replay data and parquet output

## Tests
tested locally with two different types of whole body tasks.

https://github.com/user-attachments/assets/282aafbf-ae4c-4bd4-825a-9cd2f058691d

https://github.com/user-attachments/assets/62189dd7-a503-473d-9c2c-25e6b046f066

### CODE REVIEW

**Review by xinghaozhu-nvidia — APPROVED** [review]
LGTM

File: robotic_grounding/scripts/replay_motion.py [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+            )
+            self.robot_root_wxyz = torch.tensor(
+                replay.robot_root_wxyz, dtype=torch.float32, device=device
+            )
+            self.robot_joint_pos = torch.tensor(
+                replay.robot_joint_positions, dtype=torch.float32, device=device
+            )
+
+            first_root_z = float(self.robot_root_pos[0, 2].item())
+            # Auto-detect legacy vs grounded retarget output:
```
Comment (shalin-a-jain) [inline robotic_grounding/scripts/replay_motion.py:214]: Just to confirm, is the legacy data here the examples Daniel had previously processed? Maybe we can add a todo to remove this once the ground plane changes?

**Review by shalin-a-jain — APPROVED** [review]
lgtm!

File: robotic_grounding/scripts/replay_motion.py (hunk 1 above)
Comment (huihuaNvidia2023) [reply robotic_grounding/scripts/replay_motion.py:214]: Yes, correct. I will add a todo in a follow up MR in which we need to add contact mask.
